Update to .NET 9 and use newer Dalamud conventions

emesinae opened the pull request:
# Summary

- Updated the project to target .NET 9 and C# 13.
- Updated to use ~~v11~~ [v12 Dalamud SDK](https://dalamud.dev/versions/v12/), and changed/removed some project files to follow the patterns in the SamplePlugin project
- Updated the README to include notes about environment variable required for the Dalamud SDK

A side effect: I started modifying this plugin because I was having issues where it was frequently randomly disconnecting from Intiface, but it seems like just updating everything fixed it somehow, so that's nice I guess.

Let me know if you want anything changed and I can update the PR.

emesinae committed:
build: update to .NET 9 and use newer Dalamud SDK

emesinae committed:
build: remove AssemblyInfo and related project settings

emesinae committed:
fix: delete `.targets` file

emesinae committed:
style: fix whitespace changes

digital-pet commented:
Hi! Thanks for this, if you could please restore the use of the AssemblyInfo file and project settings, those are used for automatic build versioning. Additionally, I see you removed the configuration for DalmaudPackager?

emesinae committed:
build: update Dalamud SDK to v12

emesinae committed:
build: revert a9800345, re-add AssemblyInfo

emesinae commented:
> if you could please restore the use of the AssemblyInfo file and project settings, those are used for automatic build versioning. 

I'm pretty sure [that isn't needed  any more](https://stackoverflow.com/a/42143079), but I reverted it to keep the changes simpler here. 🙂

>  I see you removed the configuration for DalmaudPackager? That tool is part of the build process that generates the zip and associated json for the custom plugin repo.

I removed the manual DLL imports, and instead changed it to use the Dalamud SDK (see the change on [line 1 of the csproj file](https://github.com/aka-tamagotchi/AetherSenseRedux/pull/20/files#diff-0060ad0409ea6997ba4f8c60a9d25b79b9f2eadc7f659d486dc7a7173e972889R1)). This is what they recommend in [the newer SampleProject](https://github.com/goatcorp/SamplePlugin/blob/master/SamplePlugin/SamplePlugin.csproj#L2)  (Discord references: [1](https://discord.com/channels/581875019861328007/653504487352303619/1318824915440373800), [2](https://discord.com/channels/581875019861328007/653504487352303619/1307036873100759081), [3](https://discord.com/channels/581875019861328007/653504487352303619/1259594937029234708)).

emesinae commented:
Also, I edited the main PR description, but I'll mention it here too: I also just updated the Dalamud SDK to use the [v12](https://dalamud.dev/versions/v12/) that was released a couple days ago.

emesinae committed:
build(dalamud): bump sdk to 12.0.2
